src/lua/compat.lua

```diff
@@ -446,6 +446,28 @@ function compat_mt.__index(_, key)
         result.current = 'old'
     end
 
+    -- Whether the effective value of the option is `new`.
+    function result.is_new(option)
+        if type(option) ~= 'table' then
+            error('usage: compat.<option_name>:is_new()')
+        end
+        if option.current == 'new' then
+            return true
+        end
+        if option.current == 'old' then
+            return false
+        end
+        return option.default == 'new'
+    end
+
+    -- Whether the effective value of the option is `old`.
+    function result.is_old(option)
+        if type(option) ~= 'table' then
+            error('usage: compat.<option_name>:is_old()')
```

Comment:
Nit: we could pass an option name to `is_old` and use option name in a message `error('usage: yaml_pretty_multiline:is_old()')`. I suppose it is more user-friendly.

I'm not insist, feel free to ignore.

Reply:
I really like the idea! Applied.
```diff
-            error('usage: compat.<option_name>:is_old()')
+            error(('usage: compat.%s:is_old()'):format(key))
```

Reply:
\*Sigh\*

This spot is fixed, but another one in `is_new` is not. Glad, this is not critical thing in this PR.

@Lord-KA, please next time **resolve** all the discussions with their owners and **do not assign** me, until all of them are resolved.

@ligurio, I guess, it's worth mentioning all the spots either via commenting the particular lines or by mentioning everything in the one comment.